src/common/mfu_daos.c

```diff
@@ -610,6 +610,290 @@ static int daos_any_error(int rank, bool local_daos_error, int flag_daos_args)
     return 0;
 }
 
+/*
+ * Get the user attributes for a container in a format similar
+ * to what daos_cont_set_prop expects.
+ * The last entry is the ACL and is conditionally set only if
+ * the user has permissions.
+ * Returns 1 on error, 0 on success.
+ */
+static int cont_get_props(daos_handle_t coh, daos_prop_t** _props)
+{
+    int             rc;
+    daos_prop_t*    props = NULL;
+    daos_prop_t*    prop_acl = NULL;
+    daos_prop_t*    props_merged = NULL;
+
+    /* Allocate space for all props except ACL. */
+    props = daos_prop_alloc(16);
+    if (props == NULL) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to allocate container properties.");
+        rc = 1;
+        goto out;
+    }
+
+    /* The order of properties MUST match the order expected by serialization. */
+    props->dpp_entries[0].dpe_type = DAOS_PROP_CO_LABEL;
+
+    props->dpp_entries[1].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
+    props->dpp_entries[2].dpe_type = DAOS_PROP_CO_LAYOUT_VER;
+    props->dpp_entries[3].dpe_type = DAOS_PROP_CO_CSUM;
+    props->dpp_entries[4].dpe_type = DAOS_PROP_CO_CSUM_CHUNK_SIZE;
+    props->dpp_entries[5].dpe_type = DAOS_PROP_CO_CSUM_SERVER_VERIFY;
+    props->dpp_entries[6].dpe_type = DAOS_PROP_CO_REDUN_FAC;
+    props->dpp_entries[7].dpe_type = DAOS_PROP_CO_REDUN_LVL;
+    props->dpp_entries[8].dpe_type = DAOS_PROP_CO_SNAPSHOT_MAX;
+    props->dpp_entries[9].dpe_type = DAOS_PROP_CO_COMPRESS;
+    props->dpp_entries[10].dpe_type = DAOS_PROP_CO_ENCRYPT;
+    props->dpp_entries[11].dpe_type = DAOS_PROP_CO_OWNER;
+    props->dpp_entries[12].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+    props->dpp_entries[13].dpe_type = DAOS_PROP_CO_DEDUP;
+    props->dpp_entries[14].dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+    props->dpp_entries[15].dpe_type = DAOS_PROP_CO_ALLOCED_OID;
+
+    /* Get all props except ACL first. */
+    rc = daos_cont_query(coh, NULL, props, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to query container: "DF_RC, DP_RC(rc));
+        rc = 1;
+        goto out;
+    }
+
+    /* Fetch the ACL separately in case user doesn't have access */
+    rc = daos_cont_get_acl(coh, &prop_acl, NULL);
+    if (rc == 0) {
+        props_merged = daos_prop_merge(props, prop_acl);
+        if (props_merged == NULL) {
```

Comment:
@daltonbohning so you are merging the all into the existing property list? Do you know if this gets appended to the list as the last entry?

Reply:
@dsikich Yes, it gets appended. I had to look at the source code for this one:
https://github.com/daos-stack/daos/blob/adb2578ac6659ff510309de63aaa10b660f65301/src/common/prop.c#L104
Basically, this is what the merge does:
```
prop_a = [A=0, B=1, C=2]
prop_b = [B=4, D=5]
merge(prop_a, prop_b) = [A=0, B=4, C=2, D=5]
```
I guess it's _**possible**_ that this implementation could change, but probably unlikely for our use case, since both property lists are distinct (no overlap).

Reply:
@daltonbohning ok, that makes sense you had to look at the source code. I looked at the header and it didn't say.